textdb/textdb-api/src/main/java/edu/uci/ics/textdb/api/common/ITuple.java

```diff
@@ -5,8 +5,16 @@
 /**
  * Created by chenli on 3/25/16.
  */
-public interface ITuple {
+public interface ITuple extends Cloneable {
     IField getField(int index);
+
     IField getField(String fieldName);
+
     List<IField> getFields();
+
+    List<Attribute> getSchema();
+
+    void addField(Attribute attribute, IField field);

```

Comment:
It's safer to make the `ITuple` immutable though it may add some cost. If an operator needs to produce another tuple based on the source tuple, it should create a new tuple instead of calling `addField` or `deleteField` to mutate the given tuple.